main.go

```diff
@@ -0,0 +1,108 @@
+package main
+
+import (
+	"database/sql"
+	"flag"
+	"fmt"
+	"html/template"
+	"log"
+	"os"
+	"time"
+
+	"github.com/caarlos0/env"
+	"github.com/ilbambino/csvomatic/parameters"
+	"github.com/ilbambino/csvomatic/stats"
+	"github.com/ilbambino/csvomatic/worker"
+	_ "github.com/lib/pq"
+)
+
+var numberWorkers int
+var csvFile string
+
+func setupArgs() {
+	flag.IntVar(&numberWorkers, "workers", 2, "Number of workers")
+	flag.StringVar(&csvFile, "input", "query_params.csv", "Path to csv with query arguments")
+	flag.Parse()
+
+}
+
+// postgresConf holds the configuration to connect to the DB
+type postgresConf struct {
+	Port     int    `env:"DB_PORT" envDefault:"5432"`
+	Host     string `env:"DB_HOST" envDefault:"localhost"`
+	User     string `env:"DB_USER" envDefault:"postgres"`
+	Password string `env:"DB_PASSWORD" envDefault:"password"`
+	Database string `env:"DB_NAME" envDefault:"homework"`
+}
+
+// getDBConnectionFromEnv constructs a Postgres connection string from env vars or from defaults
+func getDBConnectionFromEnv() (string, error) {
+
+	cfg := postgresConf{}
+	err := env.Parse(&cfg)
+	if err != nil {
+		return "", err
+	}
+
+	return fmt.Sprintf("postgres://%s:%s@%s:%d/%s?sslmode=disable", cfg.User, cfg.Password, cfg.Host, cfg.Port, cfg.Database), nil
+}
+
+func main() {
+
+	commandStart := time.Now()
+	setupArgs()
+
+	connStr, err := getDBConnectionFromEnv()
+	if err != nil {
+		log.Fatal(err)
+	}
+	db, err := sql.Open("postgres", connStr)
```

Comment:
Is this the right place to initialize the database connection? What is the downside to initializing it here (concurrency-wise)?

Reply:
From your comment I guess you refer that the same connection to the DB is shared across all the workers, and hence we can have some concurrency problem (either contention, or even worse race conditions).

But  the docs of the [sql.Open](https://golang.org/pkg/database/sql/#Open) say:
```
The returned DB is safe for concurrent use by multiple goroutines and maintains its own pool of idle connections. Thus, the Open function should be called just once. It is rarely necessary to close a DB.
```
So I think it is a correct way in the sense of sharing the DB connection and the concurrent access to it.

I don't know if it answers your question, because I might have not understood what you meant.

Reply:
Sounds good to me.  Basically what I was asking about. (And I think I learned something new!)